### PR TITLE
fix(annotations): fix type errors that prevent running bootstrap and …

### PR DESCRIPTION
**Pull request by tomwayson**
…tests

after updating to rest-js v1.6.0, a few types had changed causing TS compile errors

AFFECTS PACKAGES:
@esri/hub-annotations

ISSUES CLOSED: #47

### CODE REVIEW

File: packages/annotations/test/mocks/ago_search.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { ISearchResult } from "@esri/arcgis-rest-items";
 
-export const annoSearchResponse: ISearchResult = {
+export const annoSearchResponse = {
```
Comment (tomwayson) [inline packages/annotations/test/mocks/ago_search.ts]: NOTE: that after fixing the above issue, I then saw this:

```bash
➜  hub.js git:(master) ✗ npm test

> @esri/hub.js@1.0.1 test /Users/tom/code/hub.js
> npm run test:node && npm run test:chrome


> @esri/hub.js@1.0.1 test:node /Users/tom/code/hub.js
> jasmine --config=jasmine.json


/Users/tom/code/hub.js/node_modules/ts-node/src/index.ts:307
        throw new TSError(formatDiagnostics(diagnosticList, cwd, ts, lineOffset))
              ^
TSError: ⨯ Unable to compile TypeScript
packages/annotations/test/mocks/ago_search.ts (3,14): Type '{ query: string; total: number; start: number; num: number; nextStart: number; results: { id: str...' is not assignable to type 'ISearchResult'.
  Types of property 'results' are incompatible.
    Type '{ id: string; owner: string; created: number; modified: number; guid: null; name: string; title: ...' is not assignable to type 'IItem[]'.
      Type '{ id: string; owner: string; created: number; modified: number; guid: null; name: string; title: ...' is not assignable to type 'IItem'.
        Property 'protected' is missing in type '{ id: string; owner: string; created: number; modified: number; guid: null; name: string; title: ...'. (2322)
```

I'm worried that will affect other people, b/c I'm assuming this mock data was at one point copy/pasted from a server response. Almost seems like `ISearchResult` needs an `[index: string]: any` tacked on the end. For now, I just yanked the type from here, but that caused this error:

```bash
➜  hub.js git:(master) ✗ npm test

> @esri/hub.js@1.0.1 test /Users/tom/code/hub.js
> npm run test:node && npm run test:chrome


> @esri/hub.js@1.0.1 test:node /Users/tom/code/hub.js
> jasmine --config=jasmine.json


/Users/tom/code/hub.js/node_modules/ts-node/src/index.ts:307
        throw new TSError(formatDiagnostics(diagnosticList, cwd, ts, lineOffset))
              ^
TSError: ⨯ Unable to compile TypeScript
packages/annotations/test/mocks/ago_search.ts (15,7): Object literal's property 'guid' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (31,7): Object literal's property 'tags' implicitly has an 'any[]' type. (7018)
packages/annotations/test/mocks/ago_search.ts (34,7): Object literal's property 'documentation' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (39,7): Object literal's property 'categories' implicitly has an 'any[]' type. (7018)
packages/annotations/test/mocks/ago_search.ts (40,7): Object literal's property 'spatialReference' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (41,7): Object literal's property 'accessInformation' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (42,7): Object literal's property 'licenseInfo' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (43,7): Object literal's property 'culture' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (44,7): Object literal's property 'properties' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (47,7): Object literal's property 'proxyFilter' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (50,7): Object literal's property 'appCategories' implicitly has an 'any[]' type. (7018)
packages/annotations/test/mocks/ago_search.ts (51,7): Object literal's property 'industries' implicitly has an 'any[]' type. (7018)
packages/annotations/test/mocks/ago_search.ts (52,7): Object literal's property 'languages' implicitly has an 'any[]' type. (7018)
packages/annotations/test/mocks/ago_search.ts (53,7): Object literal's property 'largeThumbnail' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (54,7): Object literal's property 'banner' implicitly has an 'any' type. (7018)
packages/annotations/test/mocks/ago_search.ts (55,7): Object literal's property 'screenshots' implicitly has an 'any[]' type. (7018)
packages/annotations/test/mocks/ago_search.ts (62,7): Object literal's property 'groupDesignations' implicitly has an 'any' type. (7018)
```

And that is why I set `noImplicitAny: false` below. From my time developing Angular apps, I found `noImplicitAny` to be the devil.

Comment (tomwayson) [reply packages/annotations/test/mocks/ago_search.ts]: I now see that we [use `noImplicitAny: true` in rest-js](https://github.com/Esri/arcgis-rest-js/blob/0e22dba014ec661eb0650334b03f85fd2e9b0d62/tsconfig.json#L26), and disabling it project-wide seems a bit heavy handed, so I re-enabled it in ae9a7b0

**Comment by jgravois** [comment]
i finally figured out what was going on, made a few more tweaks and rebased.

File: packages/annotations/test/mocks/ago_search.ts (hunk 1 above)
Comment (jgravois) [reply packages/annotations/test/mocks/ago_search.ts]: > `Property 'protected' is missing in type` '{ id: string; owner: string; created: number..

its likely this is an example of something that TypeScript [3.0](https://blogs.msdn.microsoft.com/typescript/2018/07/30/announcing-typescript-3-0/#improved-errors-and-ux) is capable of emitting a more legible error for.

**Pull request merged**